File: r/scatterplot.r
install.packages('ggplot2')
library(ggplot2)

# Interpolate more points to make it easier to visualize
# model prediction. This is especially necessary for CART
# algorithms to properly show the nodes.
smooth_regression_line = function(test, model) {
	X_grid = seq(min(test$x1), max(test$x1), 0.1)
	y_pred = predict(mode, newdata=X_grid)
	return(list[X_grid, y_pred])
}

# Scatter plot dataset with regression line
scatterplot = function(train, test, y_pred, model, is_smooth) {
	if (is_smooth == TRUE) {
		list[X_grid, y_pred] = smooth_regression_line(test, model)
		test = list[1]
		y_pred = list[2]
	}
	
	ggplot() +
		geom_point(aes(x=train$x1, y=train$y), colour='red') +
		geom_point(aes(x=test$x1, y=test$y), colour='pink') +
		geom_line(aes(x=test$x1, y=y_pred), colour='blue') +
		ggtitle('Prediction') +
		xlab('Independent variable') +
		ylab('Dependent variable')
}

File: python/scatterplot.py
import matplotlib.pyplot as plt

# Interpolate more points to make it easier to visualize
# model prediction. This is especially necessary for CART
# algorithms to properly show the nodes.
def smooth_regression_line(X_test, model):
	X_grid = np.arange(min(X_test), max(X_test), 0.1)
	X_grid = X_grid.reshape((len(X_grid), 1))
	y_pred = model.predict(X_grid)
	return X_grid, y_pred

# Scatter plot dataset with regression line
def scatterplot(X_train, X_test, y_train, y_test, y_pred, model, is_smooth):
	if is_smooth == True:
		X_test, y_pred = smooth_regression_line(X_test, model)
		
	plt.scatter(X_train[:, 0], y_train, color='red')
	plt.scatter(X_test[:, 0], y_test, color='pink')
	plt.plot(X_test[:, 0], y_pred, color='blue')
	plt.title('Prediction')
	plt.xlabel('Independent variable')
	plt.ylabel('Dependent variable')
	plt.show()
